src/middlewared/middlewared/alert/source/ntp_status.py

```diff
@@ -0,0 +1,162 @@
+    'o': {'code': 7, 'message': 'sel_pps.peer', 'description': "PPS peer (when the prefer peer is valid)"},
+}
+
+# type - local, unicast, multicast or broadcast
+TYPE = {
+    'l': 'local',
+    'u': 'unicast',
+    'm': 'multicast',
+    'b': 'broadcast',
+    'p': 'pool'
+}
+
+FIELDS = ['remote', 'refid', 'stratum', 'type', 'when', 'poll', 'reach', 'delay', 'offset', 'jitter']
+FIELDS_CNT = len(FIELDS)
+
+
+def since_boot():
+    return time.time() - psutil.boot_time()
+
+
+def reachability(byte):
+    return '{0:08b}'.format(byte)
+
+
+def parse_peers(output):
+    peers = []
+
+    try:
+        remote = ''
+        select = ''
+        for line in output.rpartition('=====')[2].strip().splitlines():
+            """ """
+            peer = line.strip().split()
+            peer_cnt = len(peer)
+
+            """Extract peer select status and peer address"""
+            if peer_cnt in (1, FIELDS_CNT):
+                match = SELECT_STATUS_RE.match(peer[0])
+                select = match.group(0) if match else ''
+                remote = peer[0].lstrip(select)
+                if peer_cnt == 1:
+                    continue
+                peer[0] = remote
+            elif peer_cnt == FIELDS_CNT-1 and remote:
+                """Continuation line"""
+                peer.insert(0, remote)
+                remote = ''
+            else:
+                raise Exception("invalid row: '{}'".format(line))
+
+            info = dict(zip(FIELDS, peer))
+
+            info['select'] = SELECT_STATUS.get(select, '')['message']
+            info['type'] = TYPE.get(info['type'], '-')
+            info['reach'] = int(info['reach'], 8)
+
+            if info['refid'] == '.INIT.':
+                if since_boot() < STARTUP_PERIOD:
+                    break
+                else:
+                    raise Exception("still initializing after {} seconds from boot".format(since_boot()))
+
+            """Skip discarded peers """
+            if info['select'] not in ('sel_candidate', 'sel_backup', 'sel_sys.peer', 'sel_pps.peer'):
+                continue
+
+            peers.append(info)
+            remote = ''
+            select = ''
+
+    except Exception as e:
+            logger.warning("NTP status: {}".format(str(e)))
+            return []
+
+    return peers
+
+
+class NTPStatusAlertSource(AlertSource):
+    level = AlertLevel.CRITICAL
+    title = "NTP status"
+
+    schedule = IntervalSchedule(timedelta(minutes=1))
+
+    async def _get_peers(self):
+        if os.path.exists('/var/tmp/ntp/query'):
+            with open('/var/tmp/ntp/query', encoding='utf-8') as query:
+                peers = query.read()
+        else:
+            peers = (await run(['/usr/bin/ntpq', '-pwn'], encoding='utf8')).stdout
+
+        return parse_peers(peers)
+
+    async def check(self):
+        alerts = []
+
+        info = next((peer for peer in await self._get_peers() if peer.get('select', '-') in ('sel_sys.peer')), None)
```

Comment:
`('sel_sys.peer')` is a string, to make it tuple you should add a comma `('sel_sys.peer',)`

Reply:
Right... Strangely it worked though.

Reply:
Because "sel_sys.peer" in "sel_sys.peer" is True.
If you dont plan on adding more types, == "sel_sys.peer" makes more sense

Reply:
I wasn't sure if it won't need to be extended, but for now `==` should fit.